app/bridge/client.js

```diff
@@ -7,24 +7,30 @@ function ClientFactory(name, version) {
 
 const kErrors = {
   NotImplemented: 'Not Implemented.',
-  NoPromise: 'No Promise Found.'
+  NoPromise: 'No Promise Found.',
+  CurrentlyConnecting: 'Currently connecting.',
+  CurrentlyDisconnecting: 'Currently disconnecting',
+  Disconnected: 'We have been disconnected!'
 }
 
 function createNewClient(name, version) {
   /*
    * Global variables
    */
+  // request that needs to be sent when receiving the connected event
   var pendings = {};
+  // waiting for an answer
+  var waitings = {};
```

Comment:
What about renaming s/waitings/runnings ?
// requests currently running on the server and awaiting a response
var runnings = {};

Then when the view goes into background, the client code can look if there is any running requests because disconnecting itself from the server.